Hivemind/Assets/Scripts/Gathering.cs

```diff
@@ -59,23 +105,47 @@ private void Update()
         {
             case State.Idle:
                 agent.isStopped = true;
-                target = findResource();
-                if (target != null)
-                {
-                    agent.isStopped = false;
-                    nextHarvest = target.DecreaseFutureResources(CarryAmount - inventoryAmount);
-                    agent.SetDestination(target.GetPosition());
-                    state = State.MovingToResource;
-                }
+                TargetResource();
                 break;
+
             case State.Scouting:
+                if (!preparingReturn)
+                {
+                    preparingReturn = true;
+                    StartCoroutine(ReturnToBase());
+                }
+                if (!scouting)
+                {
+                    if (GameWorld.FindNearestUnknownResource(transform.position, ResourceType.Unknown) != null)
+                    {
+                        if (Vector3.Distance(transform.position, GameWorld.FindNearestUnknownResource(transform.position, ResourceType.Unknown).GetPosition()) < 2f)
```

Comment:
hier kan je een variable van maken en gebruiken in deze if statement en aan target toewijzen 2 regels lager